### DISCUSSION TITLE
Grant GitHub account contributor permissions for Azure IPAM

### DISCUSSION THREAD

**Discussion by serottez** [Azure/ipam#327] [Q&A]
Hi, I found a bug in the code and tried pushing to a new branch in the Git repository with my code fixes, but received the following error:

```
ERROR: Permission to Azure/ipam.git denied to serottez.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

I downloaded the CLA from https://opensource.microsoft.com/cla/, but don't see how this grants me permissions to contribute to this repository.

If anybody has previously contributed, please provide guidance on how to go about receiving contributor permissions.

Thank you!

**Comment by DCMattyG** [comment]
I believe there is some confusion here @serottez. You **do not**, and **should never** have direct access to a GitHub repository for which you are not an owner or a maintainer, and that goes for every repo that exists, not just Azure IPAM.

The proper way to submit bug fixes is through one of the following processes:
1. Open an Issue on this repository and we will work to address it as quickly as possible
2. Fork the repo, fix the bug and submit a Pull Request for us to review and possible accept into the repository

More info on opening an Issue can be found [here](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and information on opening Pull Requests can be found [here](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests).

I hope that helps. Looking forward to learning more about the bug you found and how we can address it as quickly as possible.
